### PR TITLE
Timo future eff

### PR DESCRIPTION
**Pull request by timotree3**
Tested it and it seems to work correctly in a 2 player no variant game. It is very disturbing though to have my efficiency be 0.45 during a game....

### CODE REVIEW

**Comment by timotree3** [comment]
Note that this is a pull request against your future-eff branch. You still have to merge that branch after merging this

File: client/src/game/rules/stats.ts [hunk 1]
```diff
@@ -151,19 +151,8 @@ export function cardsGotten(
...
-  // First, handle the case where no clues have been given yet
-  // Infinity is normal and expected in this case (on e.g. the first turn of the game)
-  // We must explicitly check for this because while e.g. "1 / 0" in JavaScript is infinity,
-  // "0 / 0" in JavaScript is NaN
-  if (potentialCluesLost === 0) {
-    return Infinity;
-  }
-
-  return currentCardsGotten / potentialCluesLost;
+export function efficiency(cards: number, clues: number): number {
```
Comment (Zamiell) [inline client/src/game/rules/stats.ts:154]: seems like a pointless function now tbh

Comment (timotree3) [reply client/src/game/rules/stats.ts:154]: true

Comment (Zamiell) [reply client/src/game/rules/stats.ts:154]: delete it and refactor?

File: client/src/game/ui/drawUI.ts [hunk 2]
```diff
@@ -1413,8 +1413,8 @@ function drawStatistics() {
     &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; <em>(number of cards played +<br />
     &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; number of cards with 1+ clues "on" them) /<br />
     &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; number of clues given or lost</em><br />
-    &nbsp; &nbsp; &nbsp; &nbsp; The first number is the efficiency needed to win the current game based<br />
-    &nbsp; &nbsp; &nbsp; &nbsp; on how well things are going so far (e.g. future required efficiency).<br />
+    &nbsp; &nbsp; &nbsp; &nbsp; The first number is the minimum amount of efficiency needed from <br />
+    &nbsp; &nbsp; &nbsp; &nbsp; this turn onward in order to win. (i.e. future required efficiency).<br />
```
Comment (Zamiell) [inline client/src/game/ui/drawUI.ts:1417]: should be `this turn onward in order to win (i.e. future required efficiency).`

**Comment by timotree3** [comment]
There is a bug in the current implementation: Future needed efficiency doesn't account for changes in pace due to critical cards getting discarded. It needs to be based on the `pace` function. not the `startingPace` function.

**Comment by timotree3** [comment]
TODO:
-[ ] fix aforementioned bug
-[ ] resolve #1810 
-[ ] add tests for all of the interesting stats functions besides `doubleDiscard`

File: client/src/game/rules/stats.test.ts [hunk 3]
```diff
@@ -1,21 +1,14 @@
...
 const defaultVariant = getVariant(DEFAULT_VARIANT_NAME);
 const blackVariant = getVariant("Black (6 Suits)");
 const clueStarvedVariant = getVariant("Clue Starved (6 Suits)");
-const cardsPerHand2Player = hand.cardsPerHand(2, false, false);
-const cardsPerHand4Player = hand.cardsPerHand(4, false, false);
-const cardsPerHand2PlayerOneExtra = hand.cardsPerHand(2, true, false);
-const cardsPerHand2PlayerOneLess = hand.cardsPerHand(2, false, true);
+const cardsPerHand2Player = 5;
+const cardsPerHand4Player = 4;
+const cardsPerHand2PlayerOneExtra = 6;
```
Comment (Zamiell) [inline client/src/game/rules/stats.test.ts:10]: how about we do `cardsPerHand2Player + 1` and `cardsPerHand2Player - 1`

**Comment by timotree3** [comment]
Should I define `startingMaxClues` in terms of `maxClues` now?

**Comment by Zamiell** [comment]
up to you

**Pull request merged**